### ISSUE TITLE
`rustls-native-certs` (without `webpki-roots` + `rustls-webpki` crates)

### ISSUE THREAD

**Issue by polarathene** [neonmoe/minreq#111]
The `https-rustls-probe` feature  extends the `https-rustls` feature. It'd be preferrable if the `rustls-native-certs` crate could be added by a feature that didn't implicitly bring in the `webpki` crates from the `https-rustls` feature.

A benefit is approx 70KB size reduction, additional reductions were also observed if `rustls` did not require the default features.

## Details

Presently building with the `https` / `https-rustls` feature bundles the trust store (`webpki` crates) into the binary. There is support for using an external trust store from the system at runtime with the `rustls-native-certs` crate (_via the `https-rustls-probe` feature_), but that currently enforces the `https-rustls` feature rather than `dep:rustls`.

In a size optimized build:
- 125KB total base size for HTTP only functionality.
- 700KB total base size with `rustls` added. If `default-features = false` (this was 30KB less).
  - `rust-native-certs` added about 20KB.
  - `https-rustls` brings in additional weight of `webpki-roots` and `rustls-webpki` crates, observed to add 70KB.
  - `Lazy` from `once_cell` can also be replaced without the crate via Rust's [`LazyLock`](https://doc.rust-lang.org/stable/std/sync/struct.LazyLock.html) (_although that's still fairly new so raises the MSRV a bit_). This only made about 300-ish bytes difference.

So presently adding `rustls` with the `rustls-native-certs` via the `https-rustls-probe` feature results in 796KB but could be approx 100KB less (_all improvements above applied_), or for just allowing the feature to work without bringing in `webpki-roots` and `rustls-webpki` crates, then 70KB less.

## Optimized build (675KB) vs standard (1.28MB)

**NOTE:** Sizes are observed with modifications/builds for [`minreq` as a dependency to `httpget`](https://github.com/cryptaliagy/httpget/issues/16).

About 500KB reduction currently. For this use-case could be improved to a 600KB reduction.

```bash
# 1.28MB (rustc 1.83.0-nightly (9ff5fc4ff 2024-10-03)):
cargo +nightly build --release --target x86_64-unknown-linux-musl

# Optimized build (slight adjustments since reported sizes above for 20KB improvement):
# 773,584 bytes (`https-rustls-probe` feature, with current webpki deps)
# 756,304 (with just the `https-rustls` feature, no `rustls-native-certs` dep)
# 675,088 (optimized, all improvements covered above, webpki deps excluded from build)
RUSTFLAGS='-Z location-detail=none -Z fmt-debug=none  -C link-arg=-fuse-ld=lld -C relocation-model=static' \
  cargo +nightly build --release \
  --target x86_64-unknown-linux-musl \
  --features https-rustls-probe \
  -Z build-std=std,panic_abort \
  -Z build-std-features=panic_immediate_abort,optimize_for_size

# With UPX (4.2.4) a little over 50% reduction, 440KB (from 773KB) vs 376KB (from 675KB)
# HTTP only 125KB => 69KB
upx --lzma -o minreq target/x86_64-unknown-linux-musl/release/minreq
```

```toml
[profile.release]
lto = true
opt-level = "z"
panic = "abort"
strip = true
```

For non-static builds, there is of course the `native-tls` feature, which when relevant can reduce the size of `httpget`/`minreq` build [down to 112KB for HTTPS (53KB via UPX)](https://github.com/cryptaliagy/httpget/issues/16#issuecomment-2418803374). However if openssl isn't already present, that's a much larger weight to introduce.

### Reference

`https-rustls-probe`: [Original feature request](https://github.com/neonmoe/minreq/issues/58) and [associated PR](https://github.com/neonmoe/minreq/pull/59).

https://github.com/neonmoe/minreq/blob/b940707fcb631d79d70bb6e1f087f2389e6e938b/Cargo.toml#L49-L50

https://github.com/neonmoe/minreq/blob/b940707fcb631d79d70bb6e1f087f2389e6e938b/src/connection.rs#L29-L37

**Comment by neonmoe** [comment]
Sorry about the late response! Good idea too, I'd imagine if you have rustls-native-certs, you wouldn't need webpki-roots. Technically this does change how the library works in a sort-of breaking way, but I can't imagine too many applications both using `https-rustls-probe` and still relying on some webpki-specific certs, so I think I'll release this as a minor version rather than postponing this until 3.0.

**Comment by neonmoe** [comment]
Implemented in 072db6c3d2a6b63ff5869dc48d9e564ff96b212e, released as part of 2.13.0. Thanks for the idea!
